static/js/topic_page/line_tile.tsx

```diff
@@ -0,0 +1,169 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering a line type tile.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React, { useEffect, useState } from "react";
+
+import { DataGroup, DataPoint, expandDataPoints } from "../chart/base";
+import { drawLineChart } from "../chart/draw";
+import { StatApiResponse } from "../shared/stat_types";
+import { getStatsVarLabel } from "../shared/stats_var_labels";
+import { CHART_HEIGHT } from "./constants";
+import { StatVarMetadata } from "./types";
+
+const SVG_CONTAINER_ELEMENT: React.RefObject<HTMLDivElement> = React.createRef();
+
+interface LineTilePropType {
+  id: string;
+  title: string;
+  placeDcid: string;
+  statVarMetadata: StatVarMetadata;
+}
+
+export function LineTile(props: LineTilePropType): JSX.Element {
+  const [rawData, setRawData] = useState<StatApiResponse | undefined>(null);
+  const [chartData, setChartData] = useState<DataGroup[] | undefined>(null);
+
+  useEffect(() => {
+    fetchData(props.statVarMetadata);
+  }, []);
+
+  useEffect(() => {
+    if (rawData) {
+      processData(rawData);
+    }
+  });
+
+  useEffect(() => {
+    if (chartData) {
+      drawChart(chartData);
+    }
+  });
+
+  return (
+    <div className="chart-container">
+      {chartData && (
+        <>
+          <div className="line-title">
+            <h4>{props.title}</h4>
+          </div>
+          <div
+            id={props.id}
+            className="svg-container"
+            ref={SVG_CONTAINER_ELEMENT}
+          ></div>
+        </>
+      )}
+    </div>
+  );
+
+  function fetchData(statVarMetaData: StatVarMetadata): void {
+    axios
+      .post(`/api/stats`, {
+        // Fetch both numerator stat vars and denominator stat vars
+        statVars: statVarMetaData.statVars.concat(statVarMetaData.denominator),
+        places: [props.placeDcid],
+      })
+      .then((resp) => {
+        setRawData(resp.data);
+      })
+      .catch(() => {
+        // TODO: add error message
+        setRawData(null);
+      });
+  }
+
+  function processData(rawData: StatApiResponse): void {
+    const trendData = rawToChart(rawData);
+    setChartData(trendData);
+  }
+
+  function drawChart(chartData: DataGroup[]): void {
+    const elem = document.getElementById(props.id);
+    elem.innerHTML = "";
+    const isCompleteLine = drawLineChart(
+      props.id,
+      elem.offsetWidth,
+      CHART_HEIGHT,
+      chartData,
+      false,
+      false,
+      props.statVarMetadata.unit
+    );
+    if (!isCompleteLine) {
+      SVG_CONTAINER_ELEMENT.current.querySelectorAll(
+        ".dotted-warning"
+      )[0].className += " d-inline";
+    }
+  }
+
+  function rawToChart(rawData: StatApiResponse): DataGroup[] {
+    // (TODO): We assume the index of numerator and denominator matches.
```

Comment:
+1 -- it's also easier to update the proto now that we're early in development..

Reply:
+1

Reply:
Changed the stat var to a message now, with "main" and "denom" field.